-------
bool succeeded
plan generated_plan
